pkg/utils/iptables.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"unsafe"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/go-logr/logr"
+)
+
+type ipTables struct {
+	path              string
+	proto             iptables.Protocol
+	hasCheck          bool
+	hasWait           bool
+	waitSupportSecond bool
+	hasRandomFully    bool
+	v1                int
+	v2                int
+	v3                int
+	mode              string // the underlying iptables operating mode, e.g. nf_tables
+	timeout           int    // time to wait for the iptables lock, default waits forever
+}
+
+// NewIPTables wraps the creation of IPTables to patch the path to the correct implementation binary.
+// It has been introduced to avoid the risk that the command doesn't work due to missing kernel modules.
+func NewIPTables(log logr.Logger, proto iptables.Protocol) (*iptables.IPTables, error) {
+	t, err := iptables.New()
+	if err != nil {
+		return nil, err
+	}
+
+	t2 := (*ipTables)(unsafe.Pointer(t))
```

Comment:
I suggest doing a PATH lookup for iptables-legacy and iptables-nft, executing the commands to check which backend is working and creating iptables.New() with iptables.Path() option, so we don't need to do this for private field access.

Reply:
Thanks for the hint! The `Path()` method is only available on the master branch, therefore I missed it.